bittensor/_neuron/text/core_validator/__init__.py

```diff
@@ -511,9 +510,12 @@ def calculate_weights(self):
                                                                            multiple=max_allowed_ratio)
         return topk_uids, topk_weights
 
-    def weights_table(self, topk_uids, topk_weights):
+    def weights_table(self, topk_uids, topk_weights, include_uids=None, num_rows: int = None):
         r""" Prints weights table given topk_uids and topk_weights.
         """
+
+        assert len(include_uids) <= num_rows, f'{len(include_uids)} > {num_rows}'  # ensure all include_uids fit in num_rows
```

Comment:
Yes, it will stop the validator, unless self.config.neuron.restart_on_failure.

I'll comment out the assert in another pull request with some other minor improvements.